### DISCUSSION TITLE
Problem with WASM in DuckDuckGo APP by nested element in view macro

### DISCUSSION THREAD

**Discussion by kwiat1990** [leptos-rs/leptos#1974] [Q&A]
Hey, I've moved some time ago to DuckDuckGo as my default browser on my Mac and it work great. Recently I ran axum workspace starter and play with it a bit. The initial example does works in all browsers but I’ve noticed an interesing bug when nesting elements in `view!` macro as follows:

```rs
/// Renders the home page of your application.
#[component]
fn HomePage() -> impl IntoView {
    // Creates a reactive value to update the button
    let (count, set_count) = create_signal(0);
    let on_click = move |_| set_count.update(|count| *count += 1);

    view! {
            <div>
                <h1>"Welcome to Leptos!"</h1>
                <button on:click=on_click>"Click Me: " {count}</button>
            </div>
    }
}
```

With this structure it stops working in DuckDuckGo browser displaying following error in the browser’s console:`Unhandled Promise Rejection: LinkError: import function wbg:__wbindgen_closure_wrapper418 must be callable`. At the same time I can’t increase counter by clicking on the button. 

I thought DuckDuckGo uses Gecko as its rendering engine like Firefox but either it’s not the case or there’s something else, which breaks the functionality in the browser because in Firefox it still does works.

Does anybody has similar experience or could elaborate how it could be fixed?

**Comment by gbj** [comment]
I can imagine two possibilities:
1. Cache settings in your DuckDuckGo browser meaning that there was a mismatch between the WASM and JS files; it's worth trying a rebuild, clearing the cache and reloading
2. DuckDuckGo introducing some extra security measures that prevent loading something

I can't imagine why this would change with your `view` macro changing.

**Reply by kwiat1990** [reply]
The routing also doesn’t work as intended. On route change the entire page gets be re-loaded. I have tried to reproduce this behaviour in other browsers but I couldn't really, so I tend to think it’s on DuckDuckGo side and we can’t make anything to fix it other than switch the browser while developing. On production it seems to work just fine because I visited a few sites, which are build with leptos and they work just fine without any of mentioned problems.
